### PR TITLE
docs: Add overview page for @ngrx/data

### LINKED ISSUE #1749: Docs: Add overview page for @ngrx/data
Similar to our other pages, `@ngrx/data` needs general doc pages.

What `@ngrx/data` is, why you might want to use it
Key concepts
Initial Usage

All this information can be ported from https://github.com/johnpapa/angular-ngrx-data

Create a new folder under `projects/ngrx.io/content/guide` named `data`.
Create an `index.md` for the overview page.
Add a section in the `projects/ngrx.io/content/navigation.json` for `Data` below the `Entity` section.

### PR DESCRIPTION
**Pull request by jtcrowson**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #1749 

## What is the new behavior?

- Adds overview section
- Describes what `@ngrx/data` is
- Key concepts
- Usage

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/data/index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    this.close();
+  }
+
+  update(hero: Hero) {
+    this.heroService.update(hero);
+  }
+}
+</code-example>
+
+In this example, we need to listen for the stream of hereos. We set the `heroes$` property to the Observable returned from the `heroeService.entities$`.  When state is changed as a result of a successful HTTP request (initiated by `getAll()`, for example), the heroes$ property will emit the latest Hero array.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/index.md]:
```suggestion
In this example, we need to listen for the stream of heroes. We set the `heroes$` property to the Observable returned from the `heroeService.entities$`.  When state is changed as a result of a successful HTTP request (initiated by `getAll()`, for example), the heroes$ property will emit the latest Hero array.
```

**Comment by jtcrowson** [comment]
@timdeschryver definitely.  I basically summarized that documentation in this PR.  Do we want to credit the original project somewhere?

File: projects/ngrx.io/content/guide/data/index.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# NgRx Data
+
+NgRx Data is an extension that offers a gentle introduction to NgRx by simplifying management of **entity data** while reducing the amount of boilerplate.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/data/index.md]:
```suggestion
NgRx Data is an extension that offers a gentle introduction to NgRx by simplifying management of **entity data** while reducing the amount of explicitness.  
```

File: projects/ngrx.io/content/guide/data/index.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+## Introduction
+
+Many applications have substantial _domain models_ with 10s or 100s of entity types.
+
+In NgRx, to create, retrieve, update, and delete (CRUD) data for every entity type is an overwhelming task. You're writing _actions_, _action-creators_, _reducers_, _effects_, _dispatchers_, and _selectors_ as well as the HTTP GET, PUT, POST, and DELETE methods _for each entity type_.
+
+In even a small model, this is a ton of repetitive code to create, maintain, and test.
+
+NgRx Data allows consumers to use NgRx while radically reducing the boilerplate necessary to manage entities with NgRx.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/data/index.md]:
```suggestion
NgRx Data allows consumers to use NgRx while radically reducing the explicitness necessary to manage entities with NgRx.
```

File: projects/ngrx.io/content/guide/data/index.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+## Key Concepts
+
+- NgRx Data can be used with NgRx to provide simplified management of entity data.
+- NgRx Data automates the creation of actions, reducers, effects, dispatchers, and selectors for each entity type.
+- NgRx Data provides default HTTP GET, PUT, POST, and DELETE methods for each entity type.
+- NgRx Data should not be used for non-entity or highly idiosyncratic data.  They are better managed with NgRx.
+
+## Defining the entities
+
+We create an `EntityMetadataMap` to tell NgRx Data about our entities.  Add a property to the set for each entity name.
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/data/index.md]: Change the references from `We` and `our` to `You` and `your`

File: projects/ngrx.io/content/guide/data/index.md (hunk 2 above)
Comment (jtcrowson) [reply projects/ngrx.io/content/guide/data/index.md]: Fixed in 5ef16b0.

File: projects/ngrx.io/content/guide/data/index.md (hunk 3 above)
Comment (jtcrowson) [reply projects/ngrx.io/content/guide/data/index.md]: Fixed in 5ef16b0.

File: projects/ngrx.io/content/guide/data/index.md (hunk 4 above)
Comment (jtcrowson) [reply projects/ngrx.io/content/guide/data/index.md]: Updated in 5ef16b0.

File: projects/ngrx.io/content/guide/data/index.md (hunk 2 above)
Comment (jtcrowson) [reply projects/ngrx.io/content/guide/data/index.md]: <insert @brandonroberts boilerplate yelling boilerplate .gif> 😂

**Pull request merged**